types/src/abi.rs

```diff
@@ -258,28 +258,41 @@ impl Type {
         gen: &GatewayABIConfig,
```

Comment:
May we have tests to support these in case of any regressions?

With bugs, it's essential to prove that the bug is fixed via a test.

Reply:
Added test as part of new commit.